src/array.cr

```diff
@@ -607,20 +641,18 @@ class Array(T)
   # a[6..10]? # => nil
```

Comment:
to make the docs more symmetrical, can we add  something like: 
```# Like `#[Range]` but returns `nil` if the Range.begin is out of range.```

to match the docs from `#[]?(start : Int, count : Int)`
```Like `#[Int, Int]` but returns `nil` if the *start* index is out of range.```